flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemTableSink.java

```diff
@@ -379,6 +390,33 @@ private Object createWriter(Context sinkContext) {
         }
     }
 
+    private void checkAllowanceOfSettingParallelism(
+            int inputParallelism, int configuredParallelism) {
+
+        if (inputParallelism != configuredParallelism) {
+            ChangelogMode mode;
+            if (bulkWriterFormat != null) {
+                mode = bulkWriterFormat.getChangelogMode();
+            } else if (serializationFormat != null) {
+                mode = serializationFormat.getChangelogMode();
+            } else {
+                throw new TableException("Can not find format factory.");
+            }
```

Comment:
Sure, how about extracting this part of code as a private method called `getChangelogModeFromFormat`

Reply:
Sounds good to me.

Reply:
I think we shouldn't use the format changelog mode, becuase it can't represent input stream changelog mode. A debezium-json can still be used for batch mode and we don't need key shuffle in this case. 

I would suggest to check whether paralleslim is configured in `DynamicTableSink#getChangelogMode`. We also have a primary key validation check in `JdbcDynamicTableSink#getChangelogMode`.

Reply:
@wuchong 
Okay, I'll 
1. Check the changlog mode in `DynamicTableSink#getChangelogMode` instead of checking in `FileSystemTableSink#Consume`. 
2. Use `requestChangelogMode`(the parameter) instead of the format changlog mode. 
3. Have a primary key check?

Is step 3 necessary?

Reply:
1. Yes.
2. Yes. 
3. We don't need to check PK.